File: CSharp/StdLib/PackageBundleSource.cs
using StdLib;
using System.Threading.Tasks;

namespace StdLib
{
    public class PackageBundleSource : PackageSource {
        public PackageBundle bundle;
        
        public PackageBundleSource(PackageBundle bundle)
        {
            this.bundle = bundle;
        }
        
        public Task<PackageBundle> getPackageBundle(PackageId[] ids, bool cachedOnly)
        {
            throw new Error("Method not implemented.");
        }
        
        public async Task<PackageBundle> getAllCached()
        {
            return this.bundle;
        }
    }
}

File: PHP/StdLib/PackageBundleSource.php
<?php

namespace StdLib\PackageBundleSource;

use StdLib\PackageManager\PackageSource;
use StdLib\PackageManager\PackageId;
use StdLib\PackageManager\PackageBundle;

class PackageBundleSource implements PackageSource {
    public $bundle;
    
    function __construct($bundle) {
        $this->bundle = $bundle;
    }
    
    function getPackageBundle($ids, $cachedOnly) {
        throw new \OneLang\Error("Method not implemented.");
    }
    
    function getAllCached() {
        return $this->bundle;
    }
}
